domain.go

```diff
@@ -676,28 +676,83 @@ func (d *VirDomain) MemoryStats(nrStats uint32, flags uint32) ([]VirDomainMemory
 	return out, nil
 }
 
-func (d *VirDomain) GetVcpus(maxInfo int32) ([]VirVcpuInfo, error) {
+// libvirt-domain.h: VIR_CPU_MAPLEN
+func virCpuMapLen(cpu uint32) C.int {
+	return C.int((cpu + 7) / 8)
+}
+
+// extractCpuMask extracts an individual cpumask from a slice of cpumasks
+// and parses it into a slice of CPU ids
+func extractCpuMask(bytesCpuMaps []byte, n, mapLen int) []uint32 {
+	const byteSize = uint(8)
+
+	// Repslice the big array to separate only mask number 'n'
+	cpuMap := bytesCpuMaps[n*mapLen : (n+1)*mapLen]
+
+	out := make([]uint32, 0)
+	for i, b := range cpuMap { // iterate over bytes of the mask
+		for j := uint(0); j < byteSize; j++ { // iterate over bits in this byte
+			if (b>>j)&0x1 == 1 {
+				out = append(out, uint32(j+uint(i)*byteSize))
+			}
+		}
+	}
+
+	return out
+}
+
+func (d *VirDomain) GetVcpus(conn *VirConnection, maxInfo int32) ([]VirVcpuInfo, [][]uint32, error) {
```

Comment:
@rgbkrk @hinesmr  I'm feeling that an extra argument is less worrying, since you always have a connection when calling this function. I'm more worried about an extra parameter returned, I could fold the CPUs mask into `VirVcpuInfo` to avoid changing callers code